packages/react-strict-dom/src/dom/runtime.js

```diff
@@ -0,0 +1,195 @@
+  hr: {
+    backgroundColor: 'black',
+    borderStyle: 'none',
+    borderWidth: 0,
+    boxSizing: 'border-box',
+    height: 1
+  },
+  img: {
+    aspectRatio: 'attr(width) / attr(height)',
+    height: 'auto',
+    maxWidth: '100%'
+  },
+  input: {
+    borderWidth: 1,
+    borderStyle: 'solid'
+  },
+  strong: {
+    fontWeight: 'bolder'
+  },
+  textarea: {
+    borderWidth: 1,
+    borderStyle: 'solid',
+    resize: 'vertical'
+  }
+});
+
+const a: StrictReactDOMAnchorProps = { style: styles.inline };
+const article: StrictReactDOMProps = { style: styles.block };
+const aside: StrictReactDOMProps = { style: styles.block };
+const b: StrictReactDOMProps = { style: styles.inline };
+const bdi: StrictReactDOMProps = { style: styles.inline };
+const bdo: StrictReactDOMProps = { style: styles.inline };
+const blockquote: StrictReactDOMProps = { style: styles.block };
+const br: StrictReactDOMProps = {};
+const button: StrictReactDOMButtonProps = {
+  style: [styles.inlineblock, styles.button],
+  type: 'button'
+};
+const code: StrictReactDOMProps = {
+  style: [styles.inline, styles.codePre]
+};
+const div: StrictReactDOMProps = { style: styles.block };
+const em: StrictReactDOMProps = { style: styles.inline };
+const fieldset: StrictReactDOMProps = { style: styles.block };
+const footer: StrictReactDOMProps = { style: styles.block };
+const form: StrictReactDOMProps = { style: styles.block };
+const heading: StrictReactDOMProps = {
+  style: [styles.block, styles.heading]
+};
+const header: StrictReactDOMProps = { style: styles.block };
+const hr: StrictReactDOMProps = {
+  style: [styles.block, styles.hr]
+};
+const i: StrictReactDOMProps = { style: styles.inline };
+const img: StrictReactDOMImageProps = { style: styles.img };
+const input: StrictReactDOMInputProps = {
+  dir: 'auto',
+  style: [styles.inlineblock, styles.input]
+};
+const label: StrictReactDOMLabelProps = { style: styles.inline };
+const li: StrictReactDOMProps = { style: styles.block };
+const main: StrictReactDOMProps = { style: styles.block };
+const nav: StrictReactDOMProps = { style: styles.block };
+const ol: StrictReactDOMProps = { style: styles.block };
+const optgroup: StrictReactDOMOptionGroupProps = {};
+const option: StrictReactDOMOptionProps = {};
+const p: StrictReactDOMProps = { style: styles.block };
+const pre: StrictReactDOMProps = {
+  style: [styles.block, styles.codePre]
+};
+const section: StrictReactDOMProps = { style: styles.block };
+const select: StrictReactDOMSelectProps = {
+  style: [styles.inlineblock]
+};
+const span: StrictReactDOMProps = { style: styles.inline };
+const strong: StrictReactDOMProps = {
+  style: [styles.inline, styles.strong]
+};
+const sub: StrictReactDOMProps = { style: styles.inline };
+const sup: StrictReactDOMProps = { style: styles.inline };
+const textarea: StrictReactDOMTextAreaProps = {
+  dir: 'auto',
+  style: [styles.inlineblock, styles.textarea]
+};
+const ul: StrictReactDOMProps = { style: styles.block };
```

Comment:
In the transformed output you're spreading the `defaultProps` but those contain the `style` prop which is invalid which is I'm guessing why you inject a `style={null}` prop before actually spreading the `stylex.props` call. If these definitions were separated into `defaultProps` and `defaultStyles` the transformed output could be simplified a bit to just something like `<div {...defaultProps.div} {...stylex.props(defaultStyles.div)} />`. I could be missing another reason why you took that approach though.

Reply:
No you've understood correctly. It was just more code in the prod runtime to do it the way you mention. Another alternative would be to do away with `defaultProps` and have only `defaultStyles`, with the few elements that have a default prop (e.g., `dir=auto`) getting that prop supplied with some logic that is replicated in both the runtime wrapper and the babel plugin. That would probably be the most performant (fewest spreads in prod and least repeated code in bundles) approach, and the maintenance cost of duplication is probably worth it given we're unlikely to be adding/removing default props in any significant way going forward.

Reply:
Pushed a diff that removes defaultProps but preserves the same result, so the transformed code looks like this instead `<div {...stylex.props(defaultStyles.div)} />` by default, and the 3 elements that have a default prop are special cased in the wrapper component and babel plugin